crates/rv-gem-types/src/dependency.rs

```diff
@@ -53,21 +53,12 @@ impl Dependency {
         Self::new(name, requirements, Some(DependencyType::Development))
     }
 
-    pub fn matches(&self, name: &str, version: &Version, allow_prerelease: bool) -> bool {
-        if self.name != name {
-            return false;
-        }
-
-        // Check prerelease logic
-        if version.is_prerelease() && !allow_prerelease && !self.requirement.is_prerelease() {
-            return false;
-        }
-
-        self.requirement.satisfied_by(version)
+    pub fn matches(&self, name: &str, version: &Version) -> bool {
+        self.name == name && version.is_prerelease() && self.requirement.satisfied_by(version)
     }
 
     pub fn matches_spec(&self, name: &str, version: &Version) -> bool {
```

Comment:
I don't understand the difference btween `matches` and `matches_spec`, maybe you could document the difference for us non-Ruby-knowers?